### ISSUE TITLE
Unexpected subject migration data

### ISSUE THREAD

**Issue by mahalakshme** [avniproject/avni-server#629]
### Acceptance critetria:
![Image](https://github.com/avniproject/avni-server/assets/10369963/efecd411-6e51-4a49-b847-dfb6f2cffbb6)

When did `select * from subject_migration where individual_id=1114469;` we see the above data.
Why the old_sync_concept_1 value saved on 30-09-2023 will be empty?
Why the new_sync_concept_1 value saved on 5-10-2023 will be empty?

### Analysis:
On 30-9-2023, a program encounter is saved, but from the code `markSubjectMigrationIfRequired` function, cant understand why the old sync concept value will be empty
On 5-10-2023, looks like program enrolment is modified - found from last_modified_date_time of program enrolment and considering that there are no changes to program_encounter. 
- But considering the decision rule for program_enrolment seems to be correct, need to find out how the new_sync_concept_value can be empty.

### Things to note:
- When the sync_concept_value of say, program_enrolment gets updated, the last_modified_data_time will not get updated
- last_modified_data_time gets updated when observations(decisions as well) gets updated

**Comment by ashusvnath** [comment]
I paired with @vinayvenu on the analysis.

* We saw 2 entries per subject_migration with `new_sync_concept_1_value = null`. This implies that the entries are coming from the system and not manual intervention.
* This is happening for both Mother and Child programs, so no program specificity
* We strongly believe this happens when enrolment is modified only. 
* There was one stretch of cases in September (~2023-09-27) but we could not find any more details
* We have setup a daily alert for `subject_migration` entries with `new_sync_concept_1_value = null` with @mahalakshme , @ashusvnath  and @vinayvenu  as recepients. The entries are filtered by last_modified_date = current_date, based on discussions between us, the alert will be changed to  include all `subject_migration` entries for which `new_sync_concept_1_value = null` everyday.
